### PR TITLE
TO Go: implement tenants CRUD

### LINKED ISSUE #2454: TO API: "tenants" endpoint returning id as string
This is a recent regression.   The `api/1.2/tenants` endpoint returns the `id` field as a string rather than an integer.   This causes the `POST .../deliveryservice` endpoint to fail.

### PR DESCRIPTION
**Pull request by dangogh**
This fixes #2454

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1341/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1364/
Test FAILed.

File: traffic_ops/traffic_ops_golang/routes.go [hunk 1]
```diff
@@ -205,6 +206,13 @@ func Routes(d ServerData) ([]Route, http.Handler, error) {
 
 		//System
 		{1.2, http.MethodGet, `system/info/?(\.json)?$`, systeminfo.Handler(d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
+
+		//Tenants
+		{1.3, http.MethodGet, `tenants/?(\.json)?$`, api.ReadHandler(tenant.GetRefType(), d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
```
Comment (mitchell852) [inline traffic_ops/traffic_ops_golang/routes.go]: I think you should change these to 1.2 because you are simply rewriting the same perl endpoints in go

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1571/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1572/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1634/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1651/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1652/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1887/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1891/
Test PASSed.

File: traffic_ops/traffic_ops_golang/routes.go [hunk 2]
```diff
@@ -282,6 +283,13 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 		{1.1, http.MethodPost, `parameterprofile/?$`, profileparameter.PostParamProfile(d.DB.DB), auth.PrivLevelOperations, Authenticated, nil},
 		{1.1, http.MethodDelete, `profileparameters/{profileId}/{parameterId}$`, api.DeleteHandler(profileparameter.GetRefType(), d.DB), auth.PrivLevelOperations, Authenticated, nil},
 
+		//Tenants
+		{1.2, http.MethodGet, `tenants/?(\.json)?$`, api.ReadHandler(tenant.GetRefType(), d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
```
Comment (mitchell852) [inline traffic_ops/traffic_ops_golang/routes.go]: i think this is supposed to be 1.1. 

we only really have 1 true 1.2 route and that is the asn route

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1892/
Test PASSed.

**Comment by mitchell852** [comment]
seems to work as expected. I tested out the CRUD of tenants using TP. I also made sure I could create a DS. maybe somebody with real Go experience can take a look at the code.

**Pull request merged**